### ISSUE TITLE
Sorting files on ReadDir

### ISSUE THREAD

**Issue by defgenx** [pkg/sftp#404]
Hello.

I read the code and the documentation but I am unable to find a way to make the lib sorts files from a directory I am currently reading. I know I can sort it myself after reading but I think it is a basic command to be able to sort. Can someone tell me if there is a way ? Also can someone tell me the default sorting applied by the library ?

Thank you.

**Comment by puellanivis** [comment]
TL;DR: You have to sort it yourself. The code is working as intended.

There is nothing in the SSH specification we follow that defines that directory listings must be sorted, nor does the standard define any way to indicate to the server that it should return results in a sorted order.

This is the same behavior as `os.File.ReadDir()`, while some operating systems might return sorted results, nothing in the POSIX standard or SSH standard we use dictates that it `MUST` be sorted. (Notably, my linux work machine doesn’t return sorted local directory listings.)

If a caller needs some kind of guarantee of a sorted order, then it is on the caller to sort the results.

**Comment by defgenx** [comment]
@puellanivis thanks for your answer.

I am closing the issue.
